src/env/ParticleGrid.h

```diff
@@ -0,0 +1,256 @@
+#pragma once
+
+#include <string>
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+
+#include "env/Common.h"
+#include "env/Particle.h"
+#include "utils/ContainerUtils.h"
+
+/**
+ * @file ParticleGrid.cpp
+ * @brief Implements the Linked cells.
+ *
+ * The following plot visualizes the benefit of Linked Cells:
+ *
+ * @image html input/images/plot.png "Performance plot" width=800px height=600px
+ */
+
+/**
+ * @brief Contains classes and structures for managing the environment of the simulation.
+ */
+namespace md::env {
+    /**
+     * @brief Structure representing a cell of the particle grid.
+     */
+    struct GridCell {
```

Comment:
I noticed that you removed the DirectSum container. It was asked in the worksheet to keep the old container and implement the new one separately. This was also a way to check how you extend the structure for containers. Or am I missing something?

Reply:
in the worksheet it was stated "should still be possible to use the existing direct sum implementation", which we interpreted as being able to use direct sum method, not that the old container must persists, hence why we deleted the old container. 
The ability to still use the direct sum method is still there, by simply setting the cell sizes to the entirety of the boundary (this is mentioned in the report)

Reply:
i should also add that the reason why we chose to delete the old container is that it was just a very basic wrapper around a vector. our new class that maintains the particles is far more comprehensive and since it still provides the exact same functionality as the old container (can add particles, cuboids and use an index to access them and iterate over particles), it would have been redundant code duplication

Reply:
I think you make a fair argument, I accept it. I do think there is a slight overhead with using LinkedCell as DirectSum but I don't expect it to be too much :)